Update abi package/test open dollar interface exporter action

jahabeebs opened the pull request:
This relates to https://github.com/open-dollar/od-sdk/issues/33

## Description

I'd like to test the forked open-dollar/interface-exporter-action github action I created as a fork of defi-wonderland's package, but to run our Interface Exporter Action I need to 1) merge to dev (what this pr is for) and 2) make sure this repo can build properly so we can run the full action.

I had to make two changes so the repo could build:
1) deleted some contract tests
2) changed BasicActions.sol to be typed properly

It would be ideal if I could test Interface Exporter Action and build the repo without having to modify smart contracts so maybe we can talk Monday about a workaround

pi0neerpat reviewed (changes requested):
Add ability to dispatch and test with our new forked version of the github action

jahabeebs commented:
@pi0neerpat I merged with main, added workflow_dispatch and added lint fixes. it won't show me trigger workflow_dispatch probably because it's not merged with dev. can I merge?
